app/services/badge_service.rb

```diff
@@ -0,0 +1,50 @@
+class BadgeService
+  RULE_SUBJECTS = ["rule_level", "rule_category", "rule_attempt"]
```

Comment:
```suggestion
  RULE_SUBJECTS = %w[rule_level rule_category rule_attempt].freeze
```
точно нужно пофризить и можно для красоты использовать `%w[]` (массив строк так рекомендуют записывать)